specification/appplatform/resource-manager/Microsoft.AppPlatform/preview/2023-08-02-preview/SpringEnvironments.json

```diff
@@ -0,0 +1,494 @@
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Patch Managed Environment": {
+            "$ref": "./examples/v2/SpringEnvironments/SpringEnvironments_Patch.json"
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AppPlatform/springEnvironments/{springEnvironmentName}/checkNameAvailability": {
+      "post": {
+        "tags": [
+          "SpringEnvironments"
+        ],
+        "summary": "Checks the resource name availability.",
+        "description": "Checks if resource name is available.",
+        "operationId": "Namespaces_CheckNameAvailability",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/SpringEnvironmentNameParameter"
+          },
+          {
+            "$ref": "#/parameters/CheckNameAvailabilityParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/CheckNameAvailabilityResponse"
+            }
+          },
+          "default": {
+            "description": "Error response.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "SpringEnvironments_CheckNameAvailability": {
+            "$ref": "./examples/v2/SpringEnvironments/SpringEnvironments_CheckNameAvailability.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "SpringEnvironmentsCollection": {
+      "description": "Collection of Spring Environments",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SpringEnvironment"
+          }
+        },
+        "nextLink": {
+          "description": "Link to next page of resources.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "SpringEnvironmentForPatchBody": {
+      "description": "An environment for hosting Azure Spring Apps only for patch body",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/SpringEnvironment"
```

Comment:
>"$ref": "#/definitions/SpringEnvironment" [](http://example.com/codeflow?start=10&length=41)

Just want to make sure that all the non-read-only properties of SpringEnvironment definition can be patched using the PATCH request? If not, define a separate definition containing only patch-able properties for Patch Request Payload.